tasks/src/lib.rs

```diff
@@ -0,0 +1,323 @@
+use std::{
+    any::Any,
+    collections::HashMap,
+    fmt::{Debug, Formatter},
+    marker::PhantomData,
+};
+
+use imap_codec::imap_types::{
+    command::{Command, CommandBody},
+    core::Tag,
+    response::{Bye, Data, Response, Status, StatusBody, StatusV2, Tagged},
+};
+use imap_flow::client::{ClientFlow, ClientFlowCommandHandle, ClientFlowEvent};
+use tag_generator::TagGenerator;
+
+pub mod tasks;
```

Comment:
What's better?
```
use
mod
code
```
Or?
```
mod
use
code
```
IMO the second one because both `use` and `code` might depend on `mod` but not vice versa.

Reply:
I like the second one, because having something like ...

```rust
use ...;

mod foo;
use foo::bar;
```

is very cohesive (and visually documenting.)

But I think I started to use the first variant due to auto-import and rustfmt ...

```rust
use ...;

mod foo;
use foo::bar;
// newline deleted by rustfmt?
mod baz;
use baz::bar;
use std::io::Write; // Auto-import. Kept by rustfmt?
```

We can try again and see how it goes.

Reply:
I meant that all `mod`s were always at the top before any `use`s. But I don't know what's work best with `cargo fmt`.

Reply:
Should work. Let's do it if you prefer.

Reply:
I feel you.

Reply:
Let's make this a separate issue and do all changes at once when we feel like it.